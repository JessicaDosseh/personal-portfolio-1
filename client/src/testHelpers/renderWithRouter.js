import React from "react";
import { Router } from "react-router-dom";
import { render } from "react-testing-library";
import { createMemoryHistory } from "history";
import { StateProvider } from "react-conflux";
import { darkContext } from "../store/contexts";
import { globalReducer } from "../store/reducers/globalReducer";

const renderWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...render(
      <StateProvider reducer={globalReducer} stateContext={darkContext}>
        <Router history={history}>{ui}</Router>
      </StateProvider>
    ),
    history
  };
};

export default renderWithRouter;
