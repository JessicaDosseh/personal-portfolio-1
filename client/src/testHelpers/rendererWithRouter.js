import React from "react";
import { Router } from "react-router-dom";
import renderer from "react-test-renderer";
import { createMemoryHistory } from "history";
import { StateProvider } from "react-conflux";
import { darkContext } from "../store/contexts";
import { globalReducer } from "../store/reducers/globalReducer";

const rendererWithRouter = (
  ui,
  {
    route = "/",
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...renderer.create(
      <StateProvider reducer={globalReducer} stateContext={darkContext}>
        <Router history={history}>{ui}</Router>
      </StateProvider>
    ),
    history
  };
};

export default rendererWithRouter;
