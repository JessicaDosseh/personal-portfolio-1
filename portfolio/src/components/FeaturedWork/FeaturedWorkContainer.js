import React from "react";
import styled, { css, keyframes } from "styled-components";
import instaSample from "../../images/instagram-macbook.png";
import tipprSample from "../../images/tippr-macbook.png";

const sampleHover = keyframes`
  0% {
    opacity: 1;
  }
  100% {
    opacity: 0.8;
  }
`;

const FeaturedWorkContainerDiv = styled.div`
  width: 100%;
  margin: 90px 0;

  ${props =>
    props.darkmode &&
    css`
      color: white;
    `}

  p {
    display: flex;
    margin: 0 5%;
  }

  div {
    width: 100%;
    margin: 70px 0;
    position: relative;

    &:hover {
      animation: ${sampleHover} 0.5s forwards;
    }

    div {
      margin: 20px 5%;
      width: 90%;
    }
  }
`;

const FeaturedWorkContainer = props => {
  return (
    <FeaturedWorkContainerDiv darkmode={props.darkmode}>
      <p>Featured Projects</p>
      <div>
        <div>
          <a href="https://github.com/nwthomas/instagram-clone">
            <img src={tipprSample} alt="tippr portfolio samples" />
          </a>
        </div>
      </div>
      <div>
        <div>
          <a href="https://github.com/nwthomas/instagram-clone">
            <img src={instaSample} alt="Instagram portfolio samples" />
          </a>
        </div>
      </div>
    </FeaturedWorkContainerDiv>
  );
};

export default FeaturedWorkContainer;
