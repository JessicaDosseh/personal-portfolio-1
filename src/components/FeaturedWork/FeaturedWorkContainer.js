import React from "react";
import styled, { css } from "styled-components";
import instaSample from "../../images/instagram-macbook.png";
import tipprSample from "../../images/tippr-macbook.png";
import fyloSample from "../../images/fylo-macbook.png";

const FeaturedWorkContainerDiv = styled.div`
  width: 100%;
  max-width: 1200px;
  margin: 60px auto 60px;

  ${props =>
    props.darkmode &&
    css`
      color: white;
    `}

  p {
    display: flex;
  }

  div {
    display: flex;
    justify-content: center;
    flex-wrap: wrap;

    @media (min-width: 1200px) {
      justify-content: space-between;
    }

    div {
      width: 100%;
      margin: 40px 0 0;

      @media (min-width: 500px) {
        max-width: 800px;
      }

      @media (min-width: 1200px) {
        max-width: 500px;
      }

      div {
        padding: 20px 0;
        width: 100%;
        opacity: 1;
        transition: 0.2s;
        display: flex;
        align-items: center;
        flex-direction: column;

        &:hover {
          opacity: 0.8;
          padding: 16px 0 24px;
        }

        h4 {
          width: 100%;
          text-align: center;
          display: block;
          letter-spacing: 4px;
          margin-top: 20px;
        }
      }
    }
  }
`;

const FeaturedWorkContainer = props => {
  return (
    <FeaturedWorkContainerDiv darkmode={props.darkmode}>
      <p>Featured Projects</p>
      <div>
        <div>
          <div>
            <a href="https://github.com/nwthomas/tippr">
              <img src={tipprSample} alt="tippr portfolio samples" />
            </a>
            <h4>TIPPR PAYMENT APP</h4>
          </div>
        </div>
        <div>
          <div>
            <a href="https://github.com/nwthomas/instagram-clone">
              <img src={instaSample} alt="Instagram portfolio samples" />
            </a>
            <h4>INSTAGRAM CLONE</h4>
          </div>
        </div>
        <div>
          <div>
            <a href="https://github.com/nwthomas/fylo-landing-page">
              <img src={fyloSample} alt="Fylo portfolio samples" />
            </a>
            <h4>FYLO LANDING PAGE</h4>
          </div>
        </div>
      </div>
    </FeaturedWorkContainerDiv>
  );
};

export default FeaturedWorkContainer;
