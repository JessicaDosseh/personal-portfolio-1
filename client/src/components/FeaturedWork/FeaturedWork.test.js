import React from "react";
import "react-testing-library/cleanup-after-each";
import "jest-dom/extend-expect";
import { renderWithRouter, rendererWithRouter } from "../../testHelpers";

import FeaturedWorkContainer from "./FeaturedWorkContainer";

describe("<FeaturedWorkContainer />", () => {
  it("should render the FeaturedWorkContainer component without crashing", () => {
    const helpers = renderWithRouter(
      <FeaturedWorkContainer darkmode={false} />
    );
  });

  it("should match the snapshot of FeaturedWorkContainer", () => {
    const tree = rendererWithRouter(<FeaturedWorkContainer darkmode={true} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("should render the component text", () => {
    const { getByText } = renderWithRouter(
      <FeaturedWorkContainer darkmode={true} />
    );
    expect(getByText(/tippr payment app/i)).toBeDefined();
    expect(getByText(/Featured Projects/i)).toBeDefined();
  });

  it("should render the projects pictures successfully", () => {
    const { getByTestId } = renderWithRouter(
      <FeaturedWorkContainer darkmode={true} />
    );
    expect(getByTestId("tipprPicture")).toBeDefined();
    expect(getByTestId("instagramPicture")).toBeDefined();
    expect(getByTestId("huddlePicture")).toBeDefined();
  });
});
