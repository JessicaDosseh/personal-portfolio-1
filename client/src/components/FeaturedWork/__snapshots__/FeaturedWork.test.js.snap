// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeaturedWorkContainer /> should match the snapshot of FeaturedWorkContainer 1`] = `
<div
  className="sc-bdVaJa ckjWSq"
>
  <p
    id="featured-work"
  >
    Featured Projects
  </p>
  <div>
    <div>
      <div>
        <a
          href="https://github.com/labs14-stampd"
        >
          <img
            alt="Stampd DApp sample"
            data-testid="stampdPicture"
            src="stampd-macbook.png"
          />
        </a>
        <h4>
          Stampd Blockchain Credentialing
        </h4>
      </div>
    </div>
    <div>
      <div>
        <a
          href="https://github.com/dustinmyers/react-conflux"
        >
          <img
            alt="Conflux state management sample"
            data-testid="confluxPicture"
            src="conflux-logo.png"
          />
        </a>
        <h4>
          Conflux State Management
        </h4>
      </div>
    </div>
    <div>
      <div>
        <a
          href="https://github.com/weight-lifting-journal-bw-team/back-end-nathan-thomas"
        >
          <img
            alt="weight lifting journal sample"
            data-testid="weightLiftingJournalPicture"
            src="weight-lifting-journal-app.png"
          />
        </a>
        <h4>
          Weight Lifting Journal
        </h4>
      </div>
    </div>
    <div>
      <div>
        <a
          href="https://github.com/nwthomas/tippr"
        >
          <img
            alt="tippr portfolio sample"
            data-testid="tipprPicture"
            src="tippr-macbook.png"
          />
        </a>
        <h4>
          tippr payment app
        </h4>
      </div>
    </div>
    <div>
      <div>
        <a
          href="https://github.com/nwthomas/instagram-clone"
        >
          <img
            alt="Instagram portfolio sample"
            data-testid="instagramPicture"
            src="instagram-macbook.png"
          />
        </a>
        <h4>
          Instagram Clone
        </h4>
      </div>
    </div>
    <div>
      <div>
        <a
          href="https://github.com/nwthomas/huddle-landing-page"
        >
          <img
            alt="Huddle portfolio sample"
            data-testid="huddlePicture"
            src="huddle-landing-page.png"
          />
        </a>
        <h4>
          Huddle Landing Page
        </h4>
      </div>
    </div>
  </div>
</div>
`;
